fix: batch-analysis payload missing empty string after parsing

xieshenzh opened the pull request:
By default, Jackson serialization will ignore empty string.

This PR keeps the empty strings in the sbom after breaking the payload of batch-analysis request.
So that, empty string will be kept as the values for some required properties.

For example: `components.externalReferences.url` is a required property of CycloneDX sbom. 
Empty string should be assigned to the property, when there is no value for it.
If the property is ignored by Jackson serialization when empty string is assigned to it, the sbom will become invalid: `CycloneDX Validation error: [$.components[15].externalReferences[0].url: is missing but it is required`

ruromero reviewed (changes requested):
You just have to change the `ObjectMapperProducer.java` and replace `NON_EMPTY` by `NON_NULL`
The instance injected in all the application is the same.

xieshenzh commented:
> You just have to change the `ObjectMapperProducer.java` and replace `NON_EMPTY` by `NON_NULL` The instance injected in all the application is the same.
> 
> Please add also a test that supports your change

Changed `ObjectMapperProducer` instead, and added the test case in `batch-sbom.json`. Thanks.

ruromero commented:
/retest

ruromero commented:
LGTM but don't merge until all checks are green
